analyses/cell-type-ewings/cell_type_annotations/SCPCS000492/SCPCL000824_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,615 @@
+        column_title_gp = grid::gpar(fontsize = 12),
+        column_names_side = "bottom",
+        column_names_gp = grid::gpar(fontsize = 10),
+        column_names_rot = 90,
+        ## Legend parameters
+        heatmap_legend_param = list(
+          title = "Jaccard index",
+          direction = "vertical",
+          legend_width = unit(1.5, "in")
+        ),
+        show_heatmap_legend = classification_method == "riggi",
+      )
+    }) |>
+  # concatenate vertically into HeatmapList object
+  purrr::reduce(ComplexHeatmap::`%v%`) |>
+  ComplexHeatmap::draw(
+    heatmap_legend_side = "right",
+    # add a margin to the heatmap so labels don't get cut off
+    padding = unit(c(2, 20, 2, 2), "mm")
+  )
+```
+
+Although not perfect, there is some consistency between classifications across all gene sets. 
+Let's classify cells based on both marker genes and gene set score and then validate those classifications. 
+
+```{r}
+classification_df <- classification_df |>
+  dplyr::mutate(
+    # first create a combined classification using all gene sets
+    geneset_classification = dplyr::case_when(
+      riggi_classification == "Tumor" & zhang_classification == "Tumor" & siligan_classification == "Tumor" ~ "Tumor",
+      riggi_classification == "Normal" & zhang_classification == "Normal" & siligan_classification == "Normal" ~ "Normal",
+      .default = "Ambiguous"
+    ),
+    # now classify based on both marker genes and gene set scores
+    combined_classification = dplyr::case_when(
+      marker_gene_classification == "Normal" & geneset_classification == "Normal" ~ "Normal",
+      marker_gene_classification == "Tumor" & geneset_classification == "Tumor" ~ "Tumor",
+      .default = "Ambiguous"
+    )
+  )
+```
+
+Below we see which cells are classified as tumor, normal, or ambiguous. 
+
+```{r}
+ggplot(classification_df, aes(x = UMAP1, y = UMAP2, color = combined_classification)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw()
+```
+
+
+## Classify tumor cells with `SingleR`
+
+Another approach for classifying tumor cells is to use `SCPCL000822` as a reference with `SingleR`. 
+We feel pretty confident in those annotations as there was clear consistency between marker gene expression, gene set scores, and copy number inference. 
+
+```{r}
+# add annotations from reference data to ref sce file
+coldata_df <- colData(ref_sce) |> 
+  as.data.frame() |> 
+  dplyr::left_join(ref_labels_df, by = c("barcodes" = "cell_barcode"))
+
+# replace coldata with updated coldata 
+colData(ref_sce) <- DataFrame(coldata_df, row.names = coldata_df$barcodes)
+```
+
+
+```{r}
+# train model using reference labels
+singler_model <- SingleR::trainSingleR(
+  ref_sce,
+  labels = ref_sce$tumor_cell_classification,
+  genes = "de",
+  # only use genes found in input sce 
+  restrict = rownames(sce)
+)
+
+# get singler results 
+singler_results <- SingleR::classifySingleR(
+  trained = singler_model,
+  test = sce,
+  fine.tune = TRUE
+)
+```
```

Comment:
I think you can probably do this in the one-step mode. 

The one thing I would suggest is that you remove the `Ambiguous` classification from the SingleR reference. That is not a "real" cell class, so we don't want to assign any cells to it!

Reply:
On the other hand, I am not sure I am comfortable with using SingleR with only 2 classes: it seems likely that it will overfit those two groups. Not sure the immediate solution here, though we could perhaps see what happens with multiple references: one with just tumor-normal, one with more cell types from normal tissue?